checkov/gitlab_ci/runner.py

```diff
@@ -39,73 +50,45 @@ def is_workflow_file(self, file_path: str) -> bool:
     def included_paths(self) -> Iterable[str]:
         return (".gitlab-ci.yml", ".gitlab-ci.yaml")
 
-    def get_images(self, file_path: str) -> set[Image]:
-        """
-        Get container images mentioned in a file
-        :param file_path: File to be inspected
-        GitLab a workflow file can have a job and services run within a container.
-
-        in the following sample file we can see a node:14.16 image:
-
-        default:
-            image:
-                name: ruby:2.6
-                entrypoint: ["/bin/bash"]
-
-            image: nginx:1.18
-
-            services:
-                - name: privateregistry/stuff/my-postgres:11.7
-                  alias: db-postgres
-                - name: redis:latest
-                - nginx:1.17
-        Source: https://docs.gitlab.com/ee/ci/docker/using_docker_images.html
-
-        :return: List of container image short ids mentioned in the file.
-        Example return value for a file with node:14.16 image: ['sha256:6a353e22ce']
-        """
+    def run(
+            self,
+            root_folder: str | None = None,
+            external_checks_dir: list[str] | None = None,
+            files: list[str] | None = None,
+            runner_filter: RunnerFilter | None = None,
+            collect_skip_comments: bool = True,
+    ) -> Report | list[Report]:
+        runner_filter = runner_filter or RunnerFilter()
+        report = super().run(root_folder=root_folder, external_checks_dir=external_checks_dir,
+                             files=files, runner_filter=runner_filter, collect_skip_comments=collect_skip_comments)
+        if runner_filter.run_image_referencer:
+            if files:
+                # 'root_folder' shouldn't be empty to remove the whole path later and only leave the shortened form
+                root_folder = os.path.split(os.path.commonprefix(files))[0]
+
+            image_report = self.check_container_image_references(
+                graph_connector=None,
+                root_path=root_folder,
+                runner_filter=runner_filter,
+                definitions=self.definitions
+            )
+
+            if image_report:
+                return [report, image_report]
+
+        return report
+
+    def extract_images(
+        self, graph_connector: DiGraph | None = None, definitions: dict[str, dict[str, Any] | list[dict[str, Any]]] | None = None
+    ) -> list[Image]:
+        images: list[Image] = []
+        if not definitions:
+            return images
+
+        for file, config in definitions.items():
+            manager = GitlabCiImageReferencerManager(supported_keys=("image", "services"),
```

Comment:
I think you don't need to pass `supported_keys` through the different methods. Why not directly create it in the provider?